src/plugins/plugins.js

```diff
@@ -0,0 +1,112 @@
+/**
+ * Utility to register plugins and common namespace for keeping reference to all plugins classes.
+ */
+import { toUpperCaseFirst } from '../helpers/string';
+import { createPriorityQueue } from '../utils/dataStructures/priorityQueue';
+import { createUniqueList } from '../utils/dataStructures/uniqueList';
+import { createUniqueQueue } from '../utils/dataStructures/uniqueQueue';
+
+const ERROR_PLUGIN_REGISTERED = pluginName => `There is already registered "${pluginName}" plugin.`;
+const ERROR_PRIORITY_REGISTERED = priority => `There is already registered plugin on priority "${priority}".`;
+const ERROR_PRIORITY_NAN = priority => `The priority "${priority}" is not a number.`;
+
+/**
+ * Stores plugins names' queue with they priorities.
+ */
+const priorityPluginsQueue = createPriorityQueue({
+  errorPriorityExists: ERROR_PRIORITY_REGISTERED,
+  errorPriorityNaN: ERROR_PRIORITY_NAN,
+});
+/**
+ * Stores plugins names' queue by registration order.
+ */
+const uniquePluginsQueue = createUniqueQueue({
+  errorItemExists: ERROR_PLUGIN_REGISTERED,
+});
+/**
+ * Stores plugins references between theirs' name and class.
+ */
+const uniquePluginsList = createUniqueList({
+  errorIdExists: ERROR_PLUGIN_REGISTERED,
+});
+
+/**
+ * Gets registered plugins' names with an appropriate order.
```

Comment:
I think it might sound a little bit different, maybe like:
```
Gets registered plugins' names in the following order:
1) Plugins registered with a defined priority attribute - in the ascending order.
2) Plugins registered without a defined priority attribute, in the registration order.
```
What do you think?

Reply:
Nice! Just a small change with an article + let's keep a comma, not the hyphen.

```
Gets registered plugins' names in the following order:
1) Plugins registered with a defined priority attribute, in ascending order.
2) Plugins registered without a defined priority attribute, in the registration order.
```